website/source/docs/acl/acl-identity-providers.html.md

```diff
@@ -0,0 +1,163 @@
+
+5. Applications can use the `consul acl logout` subcommand or the [logout API
+   endpoint](/api/acl/acl.html#logout-from-identity-provider) to destroy their
+   token when it is no longer required.
+
+## Kubernetes Identity Provider
+
+The `kubernetes` identity provider type is used to authenticate to Consul using
+a Kubernetes Service Account Token. This method of authentication makes it easy
+to introduce a Consul token into a Kubernetes Pod.
+
+To use an identity provider of this type the following are required to be
+configured:
+
+* **Kubernetes Host** - The address of the Kubernetes API. This should be an
+  address that is reachable from all Consul Servers in your datacenter.
+
+* **Kubernetes CA Certificate** - The PEM encoded CA cert for use by the TLS
+  client used to talk with the Kubernetes API. NOTE: Every line must end with a
+  newline: `\n`
+
+* **Service Account JWT** - A Service Account Token (JWT) used by the Consul
+  leader to access the [TokenReview API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#create-tokenreview-v1-authentication-k8s-io)
+  and [ServiceAccount API](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#read-serviceaccount-v1-core)
+  to validate application JWTs during login. 
+
+The following is an example RBAC configuration snippet to grant the necessary
+permissions to a service account named `consul-idp`:
+
+```yaml
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: review-tokens
+  namespace: default
+subjects:
+- kind: ServiceAccount
+  name: consul-idp
+  namespace: default
+roleRef:
+  kind: ClusterRole
+  name: system:auth-delegator
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: service-account-getter
+  namespace: default
+rules:
+- apiGroups: [""]
+  resources: ["serviceaccounts"]
+  verbs: ["get"]
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: get-service-accounts
+  namespace: default
+subjects:
+- kind: ServiceAccount
+  name: consul-idp
+  namespace: default
+roleRef:
+  kind: ClusterRole
+  name: service-account-getter
+  apiGroup: rbac.authorization.k8s.io
+```
+
+### Kubernetes Login Process
+
+1. The Service Account JWT given to the Consul leader initially accesses the
+   TokenReview API to validate the provided JWT is still valid. Kubernetes
+   should be running with `--service-account-lookup`. This is defaulted to true
+   in Kubernetes 1.7, but any versions prior should ensure the Kubernetes API
+   server is started with this setting. 
+
+2. After validating that the provided JWT is still valid, the Consul leader
+   looks for an optional annotation of `consul.hashicorp.com/service-name` on
+   the resolved service account using the ServiceAccount API.
+
+    ~> TODO: should this be "role-name" or "service-name"?
+
+3. 
+
+
+
+
+
+### Role Binding Rules
```

Comment:
The rest of this doc very much feels like a WIP. I would like to see RBRs and Login/Roles under the same section (not in the Kubernetes section) closer to the top. I think each term should have preferably one, max two, paragraphs of information.